somecode.cpp

```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+   char src[40];
+   char dest[100];
+   char dest2[100];
+
+   printf("Enter your name");
+   gets(src); 
+
+   // Copy name from one place to another
+   strcpy(dest, src); 
+ 
+   int a = 10;
+   int b = 20;
+   int c = a+b;
+   printf("Result is %d",&c);
+ 
+   strncpy(dest,src, strlen(dest)); 
+   
+   for (int i=0; i < 10; i++)
+   {
+      strncat(dest, "this is bad", strlen(dest)); 
+   }
+
+   // Some code
+   memcpy(dest2, dest, sizeof dest);
```

Comment:
[DS121708: Problematic C function detected (memcpy)](https://github.com/Microsoft/DevSkim/blob/master/guidance/DS121708.md)
There are a number of conditions in which memcpy can introduce a vulnerability (mismatched buffer sizes, null pointers, etc.). More secure alternitives perform additional validation of the source and destination buffer

*Replacement:* Use memcpy_s if possible.


Suggestions:
**[1]** +   memcpy_s(dest2, <size of dest2>,  dest,  sizeof dest);